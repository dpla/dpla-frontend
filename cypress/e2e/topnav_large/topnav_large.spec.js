import topnav from "../../fixtures/topnav.json";

describe('The big screen top nav', () => {
    
    beforeEach(() => {
        cy.visit('/about')
    })

    //big header
    it(`hasn't changed on /`, () => {
        cy.getDataCy('small-screen-header').should('not.be.visible');
        cy.getDataCy('menu-button').should('not.be.visible');
        const globalHeader = cy.getDataCy('global-header');
        globalHeader.should('be.visible')
        globalHeader.toMatchSnapshot();

    });
    topnav.forEach((destPage) => {
        // the home page doesn't have a home link,
        // so we have to skip home linking to home.
        it(`on /, ${destPage.selector} navigates to ${destPage.path}`, () => {
            cy.getDataCy('global-header').within(() => {
                cy.getDataCy(destPage.selector).click();
                cy.checkTitle(destPage.title);
            });
        });
    });

})
