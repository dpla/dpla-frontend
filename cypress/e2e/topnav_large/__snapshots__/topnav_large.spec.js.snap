exports[`The big screen top nav > hasn't changed on / #0`] = `
<div class="DesktopStyles_header__vQaoL" data-cy="global-header">
  <div
    class="DesktopStyles_linksContainer__5lBIG utils_siteMaxWidth__RoKR0"
    data-cy="navigation-user"
  >
    <ul class="DesktopStyles_links__rFwvo">
      <li><a data-cy="link-home" href="/">Home</a></li>
      <li>
        <a data-cy="browse-topics" href="/browse-by-topic">Browse by Topic</a>
      </li>
      <li>
        <a data-cy="browse-partners" href="/browse-by-partner"
          >Browse by Partner</a
        >
      </li>
      <li><a data-cy="exhibitions" href="/exhibitions">Exhibitions</a></li>
      <li>
        <a data-cy="primary-source-sets" href="/primary-source-sets"
          >Primary Source Sets</a
        >
      </li>
      <li><a data-cy="my-lists" href="/lists">My Lists</a></li>
    </ul>
    <span class="DesktopStyles_divider__w0FcZ"></span>
    <ul class="DesktopStyles_links__rFwvo DesktopStyles_secondaryLinks__MdQ_i">
      <li><a data-cy="about-dpla" href="/about">About DPLA</a></li>
      <li><a data-cy="news" href="/news">News</a></li>
    </ul>
    <span class="DesktopStyles_divider__w0FcZ"></span>
    <ul class="DesktopStyles_links__rFwvo DesktopStyles_tertiaryLinks__WBUMW">
      <li><a data-cy="dpla-pro" href="https://pro.dp.la">DPLA Pro</a></li>
    </ul>
  </div>
</div>
`;

exports[`The big screen top nav > on /, link-home navigates to / #0`] = `
Digital Public Library of America
`;

exports[`The big screen top nav > on /, browse-topics navigates to /browse-by-topic #0`] = `
Browse by Topic | DPLA
`;

exports[`The big screen top nav > on /, browse-partners navigates to /browse-by-partner #0`] = `
Browse by Partner | DPLA
`;

exports[`The big screen top nav > on /, exhibitions navigates to /exhibitions #0`] = `
Exhibitions | DPLA
`;

exports[`The big screen top nav > on /, primary-source-sets navigates to /primary-source-sets #0`] = `
Primary Source Sets | DPLA
`;

exports[`The big screen top nav > on /, my-lists navigates to /lists #0`] = `
My Lists | DPLA
`;

exports[`The big screen top nav > on /, about-dpla navigates to /about #0`] = `
About Us | DPLA
`;

exports[`The big screen top nav > on /, news navigates to /news #0`] = `
News | DPLA
`;
